products/identity/Products/Sign-In Service/Engineering Docs/Authentication Types/Service Account Auth (STS)/configuration/staging_prod_setup.md

```diff
@@ -0,0 +1,120 @@
+
+In the "App Details" view of the selected environment there will be one or more `vets-api-web-xxx` "pods", click on any of them to access that pod, then select the "Terminal" tab below the pod name. From the terminal you can enter `bundle exec rails c` to access that environment's Rails instance and perform the necessary commands.
+
+Leaving the terminal window open for an extended period will eventually cause it to lose its connection and force you to close and re-open a pod and terminal session - it is best to prepare the config sequence in a separate editor so that you can quickly input the commands in a fresh pod terminal.
+
+## Creating a Public Certificate
+
+The created Service Account Config will need to include a public certificate in order to validate the service account assertion sent by the STS client.
+
+### Creating an RSA Key
+
+First, locate the private key specified in the configuration instructions. If the new STS config is called by `vets-api`, you will use the SiS `sts_client` key, found at `Settings.sign_in.sts_client.key_path`.
+
+Places Keys are found
+
+* `Settings.sign_in.sts_client.key_path` - self-STS calls
+* other places?
+
+In your Rails console, open the key file and use it to create a new `OpenSSL::PKey::RSA` private key object.
+
+```ruby
+key_file = File.read('Settings.key_path')
+rsa_key = OpenSSL::PKey::RSA.new(key_file)
+```
+
+### Signing a Certificate
+
+Next you will create a certificate and use `rsa_key.public_key` to set its `public_key` attribute. Copy and enter the following into your Rails console, making sure to change the `cert.subject` to reflect `staging.va.gov` if you are creating a staging configuration.
+
+```ruby
+cert = OpenSSL::X509::Certificate.new
+cert.version = 2 # X509 v3
+cert.serial = OpenSSL::BN.rand(160).to_i # random serial number
+cert.subject = OpenSSL::X509::Name.parse("/C=US/O=Department of Veterans Affairs/OU=VA.gov/CN=https://va.gov") # change CN to "https://staging.va.gov" for staging
+cert.issuer = cert.subject # self-signed, so issuer is the same as the subject
+cert.public_key = rsa_key.public_key
+cert.not_before = Time.now
+cert.not_after = cert.not_before + 730 * 24 * 60 * 60 # valid for two years
+
+# sign the certificate with the private key
+cert.sign(rsa_key, OpenSSL::Digest::SHA256.new)
+```
+
+## Configuration Attributes
+
+With the certificate signed and created you can now create and save a new `ServiceAccountConfig` object. Copy the configuration creation below and replace the attributes with those from the configuration instructions.
+
+```ruby
+service_account_config = SignIn::ServiceAccountConfig.new({
+  service_account_id: SecureRandom.hex, # if the STS config already has a specified service_account_id use it instead
+  description: 'A Service Account Client',
```

Comment:
Do you think it'd be useful to describe each of the fields so you know what to put in each? Really it's just the `scopes`, `audience`, `user_attributes` that prob need additional explanation (since you described the cert above)